### PR TITLE
pythonPackages.Babel: 2.7.0 -> 2.9.0

### PR DESCRIPTION
**Pull request by cheriimoya**
Update babel as the current release is over a year old and some packages
need >=2.8.0

###### Motivation for this change

Update babel as the current release is over a year old and a couple of packages i use
need >=2.8.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/Babel/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/Babel/default.nix:21]: Please remove fetchpatch from inputs.

**Comment by SuperSandro2000** [comment]
Please do not merge master into master. That seems to break the rebase action. I think the best solution to this is if you would create a new PR based on staging and do not try to convert this PR onto another target branch.

**Comment by dasJ** [comment]
Yeah why fix the action when you can fix the user instead :+1:

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>home-assistant</li>
<li>jitsi</li>
<li>nixfmt</li>
<li>pdftk</li>
<li>python38Packages.Babel</li>
<li>python39Packages.Babel</li>
<li>shellcheck</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
jitsi:
<details>

warning: EvalError
Cannot evaluate attribute ‘jitsi’ in ‘/nix/store/70np9q2y65vpvzgqmvq3pjg43k0xdj4q-nixpkgs’.
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>home-assistant: log was empty</li>
  </ul>
</details>
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>jitsi</li>
<li>nixfmt</li>
<li>pdftk</li>
<li>python38Packages.Babel</li>
<li>python39Packages.Babel</li>
<li>shellcheck</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
jitsi:
<details>

warning: EvalError
Cannot evaluate attribute ‘jitsi’ in ‘/nix/store/69brfb3rjxxqc6z3ccrb9hny4qcjivzy-nixpkgs’.
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115403` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.Babel</li>
<li>python39Packages.Babel</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python38Packages.Babel:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/Babel/default.nix:22:5:

```
   |
22 |     maintainers = with maintainers; [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
warning: python-include-tests
Add a `checkPhase` for tests, or at least `pythonImportsCheck`.

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md
</details>
</details>

File: pkgs/development/python-modules/Babel/default.nix
Comment (mweinelt) [inline pkgs/development/python-modules/Babel/default.nix:14]:
```patch
diff --git a/pkgs/development/python-modules/Babel/default.nix b/pkgs/development/python-modules/Babel/default.nix
index 4d3c96d293d..183478396eb 100644
--- a/pkgs/development/python-modules/Babel/default.nix
+++ b/pkgs/development/python-modules/Babel/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, buildPythonPackage, fetchPypi, pytz, pytest, freezegun }:
+{ stdenv, lib, buildPythonPackage, fetchPypi, pytz, pytestCheckHook, freezegun }:
 
 buildPythonPackage rec {
   pname = "Babel";
@@ -11,7 +11,7 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ pytz ];
 
-  checkInputs = [ pytest freezegun ];
+  checkInputs = [ pytestCheckHook freezegun ];
 
   doCheck = !stdenv.isDarwin;
 
```

**Comment by mweinelt** [comment]
Whoops, that went south fast. I'm working on fixing this up.

**Review by mweinelt — APPROVED** [review]
I squashed the commits and the changes look good to me.

**Pull request merged**
